### PR TITLE
TF refactored code 

### PR DESCRIPTION
**Pull request by PacoDw**
Hello @tracypholmes, here are the changes for items 6, 8, 9 on #25.

Thank you so much!

### CODE REVIEW

File: mongodbatlas/provider.go [hunk 1]
```diff
@@ -68,27 +68,25 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 	return config.NewClient(), nil
 }
 
-func encodeStateID(values ...string) string {
+func encodeStateID(values map[string]string) string {
```
Comment (marinsalinas) [inline mongodbatlas/provider.go]: What does happen when the value has `:`? did you test it?

Comment (marinsalinas) [reply mongodbatlas/provider.go]: Please add this test case on `provider_test.go`

File: mongodbatlas/provider.go [hunk 2]
```diff
@@ -68,27 +68,25 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
...
-		encodedValues = append(encodedValues, base64.StdEncoding.EncodeToString([]byte(value)))
+	for key, value := range values {
+		encodedValues = append(encodedValues, base64.StdEncoding.EncodeToString([]byte(key+":"+value)))
 	}
 	return strings.Join(encodedValues, "-")
 }
 
-func decodeStateID(stateID string) []string {
-	decodedValues := make([]string, 0)
+func decodeStateID(stateID string) map[string]string {
```
Comment (marinsalinas) [inline mongodbatlas/provider.go]: Add test case validating `:` in this function as well.

File: website/docs/r/cloud_provider_snapshot.html.markdown [hunk 3]
```diff
@@ -35,12 +58,13 @@ resource "mongodbatlas_cloud_provider_snapshot" "test" {
...
 In addition to all arguments above, the following attributes are exported:
 
-* `id` - Unique identifier of the snapshot.
+* `snapshot_id` - Unique identifier of the snapshot.
 * `created_at` - UTC ISO 8601 formatted point in time when Atlas took the snapshot.
 * `description` - Description of the snapshot. Only present for on-demand snapshots.
 * `expires_at` - UTC ISO 8601 formatted point in time when Atlas will delete the snapshot.
 * `master_key_uuid` - Unique ID of the AWS KMS Customer Master Key used to encrypt the snapshot. Only visible for clusters using Encryption at Rest via Customer KMS.
 * `mongod_version` - Version of the MongoDB server.
+* `id` -	Unique identifier used for terraform for internal manages.
```
Comment (marinsalinas) [inline website/docs/r/cloud_provider_snapshot.html.markdown]: Please change in all references to `The Terraform's unique identifier used internally for state management.`

I think this description makes more sense.

File: mongodbatlas/provider.go (hunk 1 above)
Comment (PacoDw) [reply mongodbatlas/provider.go]: Thank you @marinsalinas let me check that 👍

File: website/docs/r/cloud_provider_snapshot.html.markdown (hunk 3 above)
Comment (PacoDw) [reply website/docs/r/cloud_provider_snapshot.html.markdown]: ok thanks @marinsalinas 👍

**Comment by marinsalinas** [comment]
@PacoDw could you fix the conflicts and then merge this PR?

**Pull request merged**